corehq/apps/case_importer/do_import.py

```diff
@@ -195,6 +196,92 @@ def submit_case_blocks(self, caseblocks):
             device_id=__name__ + ".do_import",
         )
 
+    def _parse_search_id(self, row):
+        """ Find and convert the search id in an Excel row """
+
+        # Find index of user specified search column
+        search_column = self.config.search_column
+        search_id = row[search_column] or ''
+
+        try:
+            # if the spreadsheet gives a number, strip any decimals off
+            # float(x) is more lenient in conversion from string so both
+            # are used
+            search_id = int(float(search_id))
+        except (ValueError, TypeError, OverflowError):
+            # if it's not a number that's okay too
+            pass
+
+        return _convert_field_value(search_id)
+
+    def _populate_updated_fields(self, row):
+        """
+        Returns a dict map of fields that were marked to be updated
+        due to the import. This can be then used to pass to the CaseBlock
+        to trigger updates.
+        """
+        field_map = self.field_map
+        fields_to_update = {}
+        for key in field_map:
+            try:
+                update_value = row[key]
+            except Exception:
+                continue
+
+            if 'field_name' in field_map[key]:
+                update_field_name = field_map[key]['field_name'].strip()
+            else:
+                # nothing was selected so don't add this value
+                continue
+
+            if update_field_name in RESERVED_FIELDS:
+                if update_field_name == 'parent_ref':
+                    raise exceptions.InvalidCustomFieldNameException(
+                        _('Field name "{}" is deprecated. Please use "parent_identifier" instead.'))
+                else:
+                    raise exceptions.InvalidCustomFieldNameException(
+                        _('Field name "{}" is reserved').format(update_field_name))
+
+            if isinstance(update_value, str) and update_value.strip() == SCALAR_NEVER_WAS:
+                # If we find any instances of blanks ('---'), convert them to an
+                # actual blank value without performing any data type validation.
+                # This is to be consistent with how the case export works.
+                update_value = ''
+            elif update_value is not None:
+                update_value = _convert_field_value(update_value)
+
+            fields_to_update[update_field_name] = update_value
+
+        return fields_to_update
+
+    def _create_field_map(self):
+        config = self.config
+        excel_fields = config.excel_fields
+        case_fields = config.case_fields
+        custom_fields = config.custom_fields
```

Comment:
Are these variables necessary? The method uses attributes of the `config` directly below a number of times.

Reply:
They don't look to be strictly necessary, they were there in the original function and in the interest of making minimal changes were left unchanged. It isn't clear to me why these exist (looks to be only 2 uses of 2 of them and 1 of the first) while the `config.search_` vars are used directly below.

Reply:
that makes sense, good to leave them here then